statsd/options.go

```diff
@@ -297,3 +301,45 @@ func WithTelemetryAddr(addr string) Option {
 		return nil
 	}
 }
+
+// WithoutOriginDetection disables the client origin detection.
+// This feature requires Datadog Agent 7.35.0+
```

Comment:
```suggestion
// WithoutOriginDetection disables the client origin detection.
// This feature requires Datadog Agent version >=6.35.0 && <7.0.0 or Agent's versions >=7.35.0.
```

I think we should move this to `WithOriginDetection` disabling it doesn't require any version of the agent.

Reply:
indeed, done!